src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -361,41 +375,93 @@ public void Initialize()
         {
             if (_isInitialized) //precaution -> probably not necessary, but still...
                 return;
-
+            
             _isInitialized = true;
-
-            Database.Changes.OnSystemDocumentChange += OnSystemDocumentChange;
-
+            
+            MyDatabaseRecord = LoadDatabaseRecord();
+            ConflictResolver = new ResolveConflictOnReplicationConfigurationChange(this, _log);
             InitializeOutgoingReplications();
             ConflictResolver.RunConflictResolversOnce();
+
+            Server.Cluster.DatabaseChanged += OnDatabaseRecordChange;
         }
 
-        private void InitializeOutgoingReplications()
+        private void OnDatabaseRecordChange(object sender, (string dbName, long index) t)
         {
-            ReplicationDocument = GetReplicationDocument();
 
-            if (ValidateReplicaitonSource() == false)
+            if (Server == null)
+                return;
+
+            if (string.Equals(t.dbName, Database.Name, StringComparison.OrdinalIgnoreCase) == false)
+                return;
+
+            var newRecord = LoadDatabaseRecord();
+
+            if(newRecord == null)
                 return;
 
-            if (ReplicationDocument?.Destinations == null || //precaution
-                ReplicationDocument.Destinations.Count == 0)
+            var connectionChanged = ReplicationTopology.MyConnectionChanged(newRecord.Topology, Server.NodeTag, Database.Name);
+            var conflictSolverChanged = MyDatabaseRecord.ConflictSolverConfig.ConflictResolutionChanged(newRecord.ConflictSolverConfig);
+
+            MyDatabaseRecord = newRecord;
+
+            if (connectionChanged)
```

Comment:
It will be better to handle this with a check on each outgoing connection if _that_ changed. Like we do in raft. 
See the logic in https://github.com/ayende/ravendb/blob/raft/src/Raven.Server/Rachis/Leader.cs#L105